### PR TITLE
Create MPS Simulation State from Tensors

### PR DESCRIPTION
**Pull request by 1tnguyen**
### Description

Ability to do `state::from_data` with a list of MPS tensors (could be host or device memory)

  - Tensors retrieved from the MPS simulator backend.

   - Externally-provided tensors. For Python, these could be numpy or cupy `ndarray`s.

### CODE REVIEW

File: python/runtime/cudaq/algorithms/py_state.cpp [hunk 1]
```diff
@@ -158,6 +158,70 @@ void bindPyState(py::module &mod) {
                 reinterpret_cast<std::complex<double> *>(info.ptr), info.size));
           },
           "Return a state from data.")
+      .def_static(
+          "from_data",
+          [](std::vector<py::buffer> tensors) {
```
Comment (amccaskey) [inline python/runtime/cudaq/algorithms/py_state.cpp:163]: Is `py::buffer` shared_ptr like? Should this be pass by reference?

File: python/tests/kernel/test_state_mps.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    print("random state: ", state_vec)
+    stacked_state_vec = state_vec.reshape(2, 2).transpose()
+    # Do SVD
+    U, S, Vh = np.linalg.svd(stacked_state_vec)
+    # Important: Tensor data must be in column major.
+    left_tensor = np.asfortranarray(U)
+    right_tensor = np.asfortranarray(np.dot(np.diag(S), Vh))
+    state = cudaq.State.from_data([left_tensor, right_tensor])
+    state.dump()
+    assert np.isclose(state[0], state_vec[0])
```
Comment (amccaskey) [inline python/tests/kernel/test_state_mps.py:80]: does `state[0]` automatically generate the state-vector-like representation from the MPS tensors?

Comment (1tnguyen) [reply python/tests/kernel/test_state_mps.py:80]: No, it will be delegated to the `amplitude` method.

**Review by amccaskey — APPROVED** [review]
LGTM

File: python/runtime/cudaq/algorithms/py_state.cpp (hunk 1 above)
Comment (1tnguyen) [reply python/runtime/cudaq/algorithms/py_state.cpp]: absolutely. Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1711/commits/a1f4cb93b1991d52988975d8d8b2ea094fc335f0

**Pull request merged**